File: python/strings/longestCommonPrefix.py
def longestCommonPrefix(strs):
    ans = ""
    if len(strs) == 1:
        return strs[0]
    for i in range(len(strs[0])):
        print(f"Checking for {strs[0][i]}")
        for j in range(1, len(strs)):
            if i > len(strs[j]):
                return ans
            if strs[0][i] != strs[j][i]:
                return ans
        ans += strs[0][i]
    return ans


# O(m x n)
# m -> no. of character is first string
# n -> length of strs

strs = ["ab", "a"]
print(longestCommonPrefix(strs))

File: cpp/strings/longestCommonPrefix.cpp
#include <bits/stdc++.h>
using namespace std;

string longestCommonPrefix(vector<string> &strs) {
  string ans;
  int n = strs.size();

  for (int i = 0; i < strs[0].size(); i++) {
    for (int j = 1; j < strs.size(); j++) {
      if (i >= strs[j].size()) {
        return ans;
      }
      if (strs[j][i] != strs[0][i]) {
        return ans;
      }
    }
    ans += strs[0][i];
  }
  return ans;
}

int main() {
  vector<string> strs = {"aa", "ab"};
  cout << longestCommonPrefix(strs);

  return 0;
}
